libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -185,29 +187,38 @@ private static boolean isServerModule(Module requestingModule) {
         return requestingModule.isNamed() && requestingModule.getLayer() == ModuleLayer.boot();
     }
 
-    private static Module requestingModule(Class<?> callerClass) {
+    static Module requestingModule(Class<?> callerClass) {
         if (callerClass != null) {
             Module callerModule = callerClass.getModule();
             if (systemModules.contains(callerModule) == false) {
                 // fast path
                 return callerModule;
             }
         }
-        int framesToSkip = 1  // getCallingClass (this method)
-            + 1  // the checkXxx method
-            + 1  // the runtime config method
-            + 1  // the instrumented method
-        ;
         Optional<Module> module = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE)
-            .walk(
-                s -> s.skip(framesToSkip)
-                    .map(f -> f.getDeclaringClass().getModule())
-                    .filter(m -> systemModules.contains(m) == false)
-                    .findFirst()
-            );
+            .walk(s -> findRequestingModule(s.map(f -> moduleOf(f.getDeclaringClass()))));
         return module.orElse(null);
     }
 
+    /**
+     * @throws NullPointerException if the requesting module is {@code null}
+     */
+    static Optional<Module> findRequestingModule(Stream<Module> modules) {
+        return modules.map(Objects::requireNonNull)
+            .dropWhile(not(systemModules::contains)) // Skip the entitlements runtime
```

Comment:
1. For sure, a specific test would be better. How should we assemble this list of classes? Just manually?
2. It separates the filtering logic from the actual stack walk, which allows me to unit-test the filtering. I'll add javadocs.

Reply:
1. Yes, I was thinking a small list of classes manually created as a static. I think there's only 3 or 4? Then when we add new specific entitlement versions we just add one more class to this list.
2. That makes sense.

Reply:
Note that a static list may be difficult because of the MRJAR (eg `Java23EleasticsearchEntitlementChecker` won't always be there, and can't be grabbed statically). That's why I was suggesting perhaps using the module of a known entitlement class, even the module of PolicyManager, which can be grabbed statically.

Reply:
Now seeing your comment about the tests running non-modular, another possibility would be to pass the "entitlement module" object in to PolicyManager. Then the unit tests can create a fake module for this purpose dynamically.

Reply:
Oh lol I had that and then got rid of it because I didn't need it with the explicit list.

Yes that's definitely doable. I can do that.

Reply:
Oh it just dawned on me that the runtime code I want to skip is all in one module; there are no frames in the bridge.